### ISSUE TITLE
Instrumentation / Tracing for real-time view of running goroutines related to pipelines

### ISSUE THREAD

**Issue by topherbullock** [concourse/concourse#2290]
Branching off of #2249, a useful place to start on Operator-level tooling is to instrument and display the running goroutines.

Being able to see the work in-flight for the pipelines configured on an ATC would allow the operator to have a lower-level view than logs and metrics, and provide some answers to the following questions :
- How many pipelines are still running? (#2249 )
- If there's a noisy neighbour, how do I find them? (#2249 )
- If pipelines stop triggering, how do I know why? (#1298)

To make this tooling easy-access, lets add a command (or several) to fly which allows an operator to visualize the work in-flight for specific groups of goroutines ( basically anything we use `ifrit` for to run on an interval ). Once that is in place, we could move toward the ability for operators to kill, restart, or pause certain routines 

Implementation considerations:
- should we allow users filter / search the output somehow? By pipeline? By 'type'?
- should this communicate over a more real-time protocol than APIs over http?
- are there other solutions out there already for managing and visualizing `context`s 
- should we use this opportunity to face our internal daemons and replace `ifrit` with [`context`](https://golang.org/pkg/context/) (and look into [`tomb`](https://godoc.org/gopkg.in/tomb.v2).. maybe the interactive part could be called tomb-raider? ) 
- if only we knew of [some kind of tracing tool](https://github.com/lightstep/lightstep-tracer-go)

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
